### ISSUE TITLE
Training & data generation code?

### ISSUE THREAD

**Issue by xanderchf** [seoungwugoh/RGMP#1]
Thanks for releasing the demo code! I'm wondering if you are planning to release training and the code to generate simulated data?

**Comment by bhack** [comment]
Any news on this?

**Comment by tzt101** [comment]
Hi! I try to train the network and find that the error from previous segmentation mask will influence the performance. I'm wondering the details for training if you don't mind.

**Comment by xanderchf** [comment]
FYI, I forked this repo and implemented the training code at https://github.com/xanderchf/RGMP

**Comment by seoungwugoh** [comment]
Hi, all.
Sorry for late response. 
The training code written by @xanderchf looks great !. 
However, It will be very hard to reproduce our results on paper without proper pre-training on simulation data generated from static images.
It is because DAVIS set is too small to train out network by itself and it lead to severe over-fitting even though encoder is initialized by ImageNet training. (Heavy data augmentation is also recommended.)

Unfortunately, we have no plan to release our training code as the code is quite messy and I am too busy for other projects for now.

**Comment by sycztrf** [comment]
Hi! Can you tell me how to use the 'bptt_hsm' function in the train.py file? @xanderchf

**Comment by dwqy11** [comment]
Hi！ How do you deal with different length of videos in a mini-batch? In your train.py file, you used batch_size=1 when training, and if I revised it to more than 1, the error about length of video is not same shown up. @xanderchf

**Comment by seoungwugoh** [comment]
> Hi！ How do you deal with different length of videos in a mini-batch? In your train.py file, you used batch_size=1 when training, and if I revised it to more than 1, the error about length of video is not same shown up. @xanderchf

In our implementation, we make sure that each training sample has same temporal length by sampling a fixed number frames from video clips.

**Comment by dwqy11** [comment]
> > Hi！ How do you deal with different length of videos in a mini-batch? In your train.py file, you used batch_size=1 when training, and if I revised it to more than 1, the error about length of video is not same shown up. @xanderchf
> 
> In our implementation, we make sure that each training sample has same temporal length by sampling a fixed number frames from video clips.

Got it ! Thanks!

**Comment by XiandaGuo** [comment]
> FYI, I forked this repo and implemented my own training code at https://github.com/xanderchf/RGMP

Hi!Thanks for releasing your code!  I am wondering why do not you use bptt_hsm in train.py? what is the meaning of ntokens in line 134 of train.py? looking forward your replay!Thank you very much

**Comment by XiandaGuo** [comment]
> Hi! Can you tell me how to use the 'bptt_hsm' function in the train.py file? @xanderchf

Hi! I want to know if you have know how to use  the 'bptt_hsm' function in the train.py file? thanks

**Comment by Starboy-at-earth** [comment]
Why do the authors always keep their training code under the desk? Maybe it is the key to the implementation of the proposed network, rather than the fancy structure.

**Comment by zgf6781882** [comment]
> Why do the authors always keep their training code under the desk? Maybe it is the key to the implementation of the proposed network, rather than the fancy structure.

What you said is excellent!

**Comment by seoungwugoh** [comment]
@Starboy-at-earth @zgf6781882 
This work was done during an internship at company. And, I am not able to make available the entire source code other than demo scripts due to potential conflicts with the company's intellectual properties. There is no secret recipe for this method to work. I will be happy to help if you have any difficulty in reproducing our results.

**Comment by zgf6781882** [comment]
> @Starboy-at-earth @zgf6781882
> This work was done during an internship at company. And, I am not able to make available the entire source code other than demo scripts due to potential conflicts with the company's intellectual properties. There is no secret recipe for this method to work. I will be happy to help if you have any difficulty in reproducing our results.

Thank you for your help and contribution to this field. It is expected that one day a paper with complete code can be published, which will bring great help to beginners.
